satpy/readers/_id_types.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Classes used as DataID element types."""
+
+import numbers
+from collections import namedtuple
+
+try:
+    zlklass = namedtuple("ZLevel", "value units", defaults=('hPa',))
+except NameError:  # python 3.6
+    zlklass = namedtuple("ZLevel", "min central max unit")
+    zlklass.__new__.__defaults__ = ('hPa',)
+
+
+class ZLevel(zlklass):
+    """Container for vertical level information in a DataID."""
+
+    @classmethod
+    def convert(cls, zlevel):
+        """Convert `zlevel` to this type if possible."""
+        if isinstance(zlevel, (tuple, list)):
+            return cls(*zlevel)
+        elif isinstance(zlevel, numbers.Number):
+            return cls(zlevel)
+        return zlevel
+
+    def __eq__(self, other):
+        """Return if two levels are the same.
+
+        Args:
+            other (ZLevel, tuple, list, or scalar): Another ZLevel object, a
+                scalar level value, or a tuple/list with either a scalar or
+                (value, units_str).
+
+        Return:
+            True if other is a scalar and equals this value or if other is
+            a tuple equal to self, False otherwise.
+
+        """
+        if other is None:
+            return False
+        is_scalar = isinstance(other, numbers.Number)
+        is_single_seq = isinstance(other, (tuple, list)) and len(other) == 1
+        if is_scalar or is_single_seq:
+            return self == self.convert(other)
+        return super().__eq__(other)
+
+    def __ne__(self, other):
+        """Return the opposite of `__eq__`."""
+        return not self == other
+
+    def __lt__(self, other):
+        """Compare to another level."""
+        if other is None:
+            return False
+        # compare using units first
+        return self[::-1].__lt__(other[::-1])
```

Comment:
So if you have one being in "Pa" and the other one in "hPa", the value in Pascals will be considered bigger???

I think we have to be very careful here. This probably needs a full fledged unit lib to be done properly. So could be just error if the units are different?

Reply:
Good point. However, the main concern here was that listing a series of DataIDs gets them in a consistent order. We can't make it fail/error if the units are different because you'd never be able to list two DataIDs that have different properties (very likely in a grib file).